torch/tensor.py

```diff
@@ -647,25 +647,36 @@ def align_to(self, *names):
             [name for name in names if not is_ellipsis(name)],
             ellipsis_idx)
 
-    def unflatten(self, dim, namedshape):
-        r"""Unflattens the named dimension :attr:`dim`, viewing it in the shape
-        specified by :attr:`namedshape`.
+    def unflatten(self, dim, sizes):
+        r"""Expands the dimension :attr:`dim` of the :attr:`self` tensor over multiple dimensions
+        of sizes given by :attr:`sizes`.
+
+        * :attr:`sizes` is the new shape of the unflattened dimension and it can be a `namedshape` 
+          (iterable of ``(name, size)`` tuples) if :attr:`self` is `NamedTensor` or a `tuple` of
+          ints as well as `torch.Size`. The total number of elements in sizes must match the number 
+          of elements in the original dim being unflattened.
 
         Arguments:
-            namedshape: (iterable of ``(name, size)`` tuples).
+            dim (Union[str, int]): Dimension to unflatten
+            sizes (Union[Iterable[Tuple[str, int]], torch.Size]): New shape of the unflattened dimension
```

Comment:
Nit: prefer `int` to come first in dim, and `torch.Size` to come first in `sizes`. Most people won't be using this for the named tensor functionality.